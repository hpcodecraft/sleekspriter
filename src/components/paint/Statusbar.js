import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import { t } from "../../utils";
import { Hotkeys } from "../../classes";

const gridHotkey = Hotkeys.bindings.paint[8].key;
const undoHotkey = Hotkeys.bindings.paint[10].key;
const redoHotkey = Hotkeys.bindings.paint[11].key;

const Statusbar = props => {
  const gridButtonClasses = classnames({
    tiny: true,
    transparent: true,
    active: props.grid,
  });
  const historyButtonClasses = classnames({
    tiny: true,
  });

  return (
    <div className="bar">
      <span id="StatusBarCursorX">X: 0</span>
      <span id="StatusBarCursorY">Y: 0</span>
      <div id="StatusBarColor" style={{ background: "transparent" }} />
      <span id="StatusBarColorString" className="statusbar-info">
        {t("transparent")}
      </span>
      <span className="statusbar-info">
        {t("Frame")} {props.frame}, {props.pixelCount} {t("pixels")}
      </span>
      <span className="statusbar-info">
        {t("Zoom Ã—")}
        {props.zoom}
      </span>
      <div id="StatusBarButtons">
        <button
          id="historyUndo"
          className={historyButtonClasses}
          onClick={props.undo}
          disabled={!props.canUndo}
          title={t("Undo", { key: undoHotkey })}>
          <i className="flaticon-back-arrow" />
        </button>

        <button
          id="historyRedo"
          className={historyButtonClasses}
          onClick={props.redo}
          disabled={!props.canRedo}
          title={t("Redo", { key: redoHotkey })}>
          <i className="flaticon-arrow" />
        </button>

        <button
          id="toggleGrid"
          className={gridButtonClasses}
          onClick={props.gridToggle}
          title={t("Toggle grid", { key: gridHotkey })}>
          <i className="flaticon-3x3" />
        </button>
      </div>
    </div>
  );
};

Statusbar.propTypes = {
  grid: PropTypes.bool.isRequired,
  gridToggle: PropTypes.func.isRequired,
  frame: PropTypes.number.isRequired,
  pixelCount: PropTypes.number.isRequired,
  zoom: PropTypes.number.isRequired,
  undo: PropTypes.func.isRequired,
  canUndo: PropTypes.bool.isRequired,
  redo: PropTypes.func.isRequired,
  canRedo: PropTypes.bool.isRequired,
};

export default Statusbar;
