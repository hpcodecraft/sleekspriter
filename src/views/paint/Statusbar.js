import React from "react";
import classnames from "classnames";

const Statusbar = (props) => {

  const gridButtonClasses = classnames({
      tiny: true,
      transparent: true,
      active: props.grid,
    }),

    historyButtonClasses = classnames({
      tiny: true,
    });

  return (
    <div className="bar">
      <span id="StatusBarCursorX">X: 0</span>
      <span id="StatusBarCursorY">Y: 0</span>
      <div id="StatusBarColor" style={{background: "transparent"}}></div>
      <span id="StatusBarColorString">transparent</span>
      <span>Frame x, x pixels</span>
      &nbsp;
      <span>Zoom &times;{props.zoom}</span>
      <div id="StatusBarButtons">

        <button id="historyUndo" className={historyButtonClasses}>
          <i className="flaticon-back-arrow"></i>
        </button>

        <button id="historyRedo" className={historyButtonClasses}>
          <i className="flaticon-arrow"></i>
        </button>

        <button id="toggleGrid" className={gridButtonClasses} onClick={props.toggleGrid}>
          <i className="flaticon-3x3"></i>
        </button>
      </div>
    </div>
  );
};

export default Statusbar;
