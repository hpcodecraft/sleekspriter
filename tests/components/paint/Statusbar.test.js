import Statusbar from "components/paint/Statusbar";

describe("Statusbar", () => {
  let props, jsx, wrapper;
  beforeEach(() => {
    props = {
      grid: true,
      gridToggle: jest.fn(),
      frame: 1,
      pixelCount: 0,
      zoom: 10,
      undo: jest.fn(),
      canUndo: true,
      redo: jest.fn(),
      canRedo: false,
    };
    jsx = <Statusbar {...props} />;
    wrapper = shallow(jsx);
  });

  it("should render correctly", () => {
    expect(render(jsx)).toMatchSnapshot();
  });

  it("fires gridToggle handler on grid button click", () => {
    wrapper.find("#toggleGrid").simulate("click");
    expect(props.gridToggle).toHaveBeenCalledTimes(1);
  });

  it("fires undo handler on undo button click", () => {
    wrapper.find("#historyUndo").simulate("click");
    expect(props.undo).toHaveBeenCalledTimes(1);
  });

  it("fires redo handler on redo button click", () => {
    wrapper.find("#historyRedo").simulate("click");
    expect(props.redo).toHaveBeenCalledTimes(1);
  });
});
