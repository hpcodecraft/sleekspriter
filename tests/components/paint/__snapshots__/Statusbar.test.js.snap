// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statusbar should render correctly 1`] = `"<div class=\\"bar\\"><span id=\\"StatusBarCursorX\\">X: 0</span><span id=\\"StatusBarCursorY\\">Y: 0</span><div id=\\"StatusBarColor\\" style=\\"background:transparent\\"></div><span id=\\"StatusBarColorString\\" class=\\"statusbar-info\\">transparent</span><span class=\\"statusbar-info\\">Frame 1, 0 pixels</span><span class=\\"statusbar-info\\">Zoom Ã—10</span><div id=\\"StatusBarButtons\\"><button id=\\"historyUndo\\" class=\\"tiny\\" title=\\"Undo\\"><i class=\\"flaticon-back-arrow\\"></i></button><button id=\\"historyRedo\\" class=\\"tiny\\" disabled=\\"\\" title=\\"Redo\\"><i class=\\"flaticon-arrow\\"></i></button><button id=\\"toggleGrid\\" class=\\"tiny transparent active\\" title=\\"Toggle grid\\"><i class=\\"flaticon-3x3\\"></i></button></div></div>"`;
